### PR TITLE
9734 improve auth coverage

### PR DESCRIPTION
**Pull request by essamhassan**
auth: adds tests for uncovered auth funcs

Coverage 68.5% -> 76.5%

Issue #9734

Please read https://github.com/etcd-io/etcd/blob/master/CONTRIBUTING.md#contribution-flow.

### CODE REVIEW

File: auth/store_test.go [hunk 1]
```diff
@@ -167,6 +167,18 @@ func TestUserAdd(t *testing.T) {
...
 
+func TestRecover(t *testing.T) {
+	as, tearDown := setupAuthStore(t)
+	defer tearDown(t)
+
+	as.enabled = false
+	as.Recover(as.be)
+
+	if !as.IsAuthEnabled() {
+		t.Fatal("expected authEnabled got ", as.IsAuthEnabled())
```
Comment (gyuho) [inline auth/store_test.go]: `t.Fatalf("... %v",`?

File: auth/store_test.go [hunk 2]
```diff
@@ -279,6 +291,75 @@ func TestUserGrant(t *testing.T) {
...
+
+	// grants a role to the user
+	_, err := as.UserGrantRole(&pb.AuthUserGrantRoleRequest{User: "foo", Role: "role-test"})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// checks role reflects correctly
+	hr := as.HasRole("foo", "role-test")
+
```
Comment (gyuho) [inline auth/store_test.go]: Remove this blank line?

File: auth/store_test.go [hunk 3]
```diff
@@ -279,6 +291,75 @@ func TestUserGrant(t *testing.T) {
...
+		PermType: authpb.WRITE,
+		Key:      []byte("Keys"),
+		RangeEnd: []byte("RangeEnd"),
+	}
+
+	_, err = as.RoleGrantPermission(&pb.AuthRoleGrantPermissionRequest{
+		Name: "role-test-1",
+		Perm: perm,
+	})
+
```
Comment (gyuho) [inline auth/store_test.go]: Remove this blank line?

**Comment by xiang90** [comment]
lgtm

**Pull request merged**